### PR TITLE
Application Layer: Send Message

### PR DESCRIPTION
**Pull request by Sajjon**
PR for [JIRA 1118](https://radixdlt.atlassian.net/browse/RLAU-1118) Introducing `MessageSending` protocol as part of the RadixApplicationClient

```swift
public protocol MessageSending {
    func sendMessage(_ message: SendMessageAction) -> Completable
}
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,10 +22,10 @@ This is a **sneak peak** of the **coming** Application Layer API
 
 ### Radix Application
 
-The `RadixApplicationClient` is the API layer this library exposes to you as a client developer. It will allow you to create and transfer tokens and also to fetch balances for a certain Radix address.
+The `RadixApplicationClient` is the API layer this library exposes to you as a client developer. It allows you to create & transfer tokens, fetch balances and send messages.
```
Comment (laszlourszuly) [inline README.md]: Highly opinionated nitpicking: I would have preferred the `and` over the ampersand.

File: Source/Stack/0. Layer 7 - APPLICATION (Transactions, Balances)/MessageSending/MessageSending.swift [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+{
+    // swiftlint:enable opening_brace
+    
+    func sendMessage(_ message: SendMessageAction) -> CompletableWanted {
+        return sendMessage(message, readers: [ /* No extra reader */ ])
+    }
+    
+    func sendMessage(_ message: SendMessageAction, readers: [Ownable]) -> CompletableWanted {
+        
+        log.info("\(message.sender) send message to: \(message.recipient)")
```
Comment (laszlourszuly) [inline Source/Stack/0. Layer 7 - APPLICATION (Transactions, Balances)/MessageSending/MessageSending.swift]: Probably safe, but doesn't hurt asking: Are risking to leak sensitive information here? What does `message.sender` and `message.recipient` contain?

File: Tests/TestCases/Radix/0. Layer 7 - APPLICATION (Transactions, Balances)/SendMessageTests.swift [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        // WHEN: I send a non empty message without encryption
+        let request = application.sendMessage("Hey Bob, this is plain text", to: bob, encrypt: false)
+        
+        XCTAssertTrue(
+            // THEN: I see that action completes successfully
+            request.blockingWasSuccessfull(timeout: RxTimeInterval.enoughForPOW)
+        )
+    }
+    
+    // This ought to work in Radix Core, but does not, awaiting fix: https://radixdlt.atlassian.net/browse/RLAU-1342
```
Comment (laszlourszuly) [inline Tests/TestCases/Radix/0. Layer 7 - APPLICATION (Transactions, Balances)/SendMessageTests.swift]: Saved by this comment :)

File: Source/Stack/0. Layer 7 - APPLICATION (Transactions, Balances)/Transacting/Transacting.swift [hunk 4]
```diff
@@ -32,31 +32,24 @@ where
 {
     // swiftlint:enable opening_brace
     
-    func transfer(tokens transferTokenAction: TransferTokenAction) -> CompletableWanted {
+    func transfer(tokens transfer: TransferTokenAction) -> CompletableWanted {
```
Comment (laszlourszuly) [inline Source/Stack/0. Layer 7 - APPLICATION (Transactions, Balances)/Transacting/Transacting.swift]: Did we rename `transferTokenAction` to `transfer` or rather change the type of `token`?
If renaming: But why? I personally found the previous name more descriptive, yet not too long,
Else: OK.

File: Source/Stack/0. Layer 7 - APPLICATION (Transactions, Balances)/MessageSending/MessageSending.swift (hunk 2 above)
Comment (Sajjon) [reply Source/Stack/0. Layer 7 - APPLICATION (Transactions, Balances)/MessageSending/MessageSending.swift]: Those are `Address`es, good that you asked! I can remove it as well.

File: Source/Stack/0. Layer 7 - APPLICATION (Transactions, Balances)/Transacting/Transacting.swift (hunk 4 above)
Comment (Sajjon) [reply Source/Stack/0. Layer 7 - APPLICATION (Transactions, Balances)/Transacting/Transacting.swift]: Only visible within the scope of the function, i.e. the implementation. The implementations does not care so much about it being an action, but rather it's some transfer/transaction/payment model, containing the necessary data to perform this transfer. In that context I felt `transferTokenAction` did not provide any more meaningful information than `transfer`.

**Pull request merged**